locksmith/src/websub/jobs/keys.ts

```diff
@@ -102,3 +111,157 @@ export async function notifyOfKeys(hooks: Hook[]) {
 
   await Promise.allSettled(tasks)
 }
+
+export async function notifyKeyExpiration() {
+  const web3Service = new Web3Service(networks)
+
+  const now = new Date()
+
+  const end = new Date(now.getTime())
+  end.setDate(now.getDate() + 1)
+
+  await Promise.allSettled(
+    // get expiring keys for every network
+    Object.keys(networks).map(async (networkId: string) => {
+      const subgraph = new SubgraphService()
+      // get keys from now to 24h in the future
+      const keys = await subgraph.keys(
+        {
+          first: 1000, // more than 1000 limit? need to handle it
+          where: {
+            expiration_gt: now.getTime().toString(),
+            expiration_lt: end.getTime().toString(),
+          },
+        },
+        {
+          networks: [Number(networkId)],
+        }
+      )
+
+      await Promise.allSettled([
+        keys?.map(async (key: any) => {
+          const lockName = key?.lock?.name ?? ''
+          const lockAddress = Normalizer.ethereumAddress(key.lock.address)
+          const ownerAddress = Normalizer.ethereumAddress(key.owner)
+          const tokenAddress = key?.lock?.tokenAddress ?? ''
+          const keyId = key?.tokenId ?? ''
+
+          const userTokenMetadataRecord = await UserTokenMetadata.findOne({
+            where: {
+              tokenAddress: lockAddress,
+              userAddress: ownerAddress,
+            },
+          })
+
+          const protectedData = Normalizer.toLowerCaseKeys({
```

Comment:
Maybe better to write this in function that does this since we seem to use that in multiple places. Use `UserTokenOperations`